lib/reverse_sentence.rb

```diff
@@ -1,6 +1,35 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if my_sentence == nil
+    return nil
+  end
+  
+  regex = /[A-Za-z']+[!\?\.,]?| +/
+  a = my_sentence.scan(regex) #O(n)
```

Comment:
Here you are making an array of all the words.  You were asked to do the reversal in place.  Instead of swapping the words in the `a` array, swapping individual characters in the original string?